packages/cletus/src/operations/internet.tsx

```diff
@@ -372,3 +375,170 @@ export const web_api_call = operationOf<
     showInput, showOutput
   ),
 });
+
+// ============================================================================
+// Web Download
+// ============================================================================
+
+export const web_download = operationOf<
+  { url: string; target: string },
+  { path: string; size: number; extension: string }
+>({
+  mode: 'create',
+  signature: 'web_download(url: string, target: string)',
+  status: (input) => `Downloading: ${abbreviate(input.url, 35)}`,
+  analyze: async ({ input }) => {
+    // Validate URL
+    try {
+      new URL(input.url);
+    } catch {
+      return {
+        analysis: `This would fail - "${input.url}" is not a valid URL.`,
+        doable: false,
+      };
+    }
+
+    // Validate target path
+    if (!input.target || input.target.trim() === '') {
+      return {
+        analysis: 'This would fail - target path cannot be empty.',
+        doable: false,
+      };
+    }
+
+    // Check if target path is absolute or relative
+    const targetPath = path.isAbsolute(input.target) ? input.target : path.resolve(input.target);
+    
+    // Check if parent directory exists
+    const parentDir = path.dirname(targetPath);
+    try {
+      await fs.promises.access(parentDir, fs.constants.W_OK);
+    } catch {
+      return {
+        analysis: `This would fail - parent directory "${parentDir}" does not exist or is not writable.`,
+        doable: false,
+      };
+    }
+
+    // Check if file already exists at the target path
+    try {
+      await fs.promises.access(targetPath, fs.constants.F_OK);
+      return {
+        analysis: `This would fail - file already exists at "${targetPath}". Please specify a different target path.`,
+        doable: false,
+      };
+    } catch {
+      // File doesn't exist, which is what we want
+    }
+
+    return {
+      analysis: `This will download the file from "${input.url}" and save it to "${targetPath}"${!path.extname(targetPath) ? ' (extension will be auto-detected if needed)' : ''}.`,
+      doable: true,
+    };
+  },
+  do: async ({ input }, { signal }) => {
+    // Check if operation was cancelled
+    if (signal?.aborted) {
+      throw new Error('Operation cancelled');
+    }
+
+    // Resolve target path
+    const targetPath = path.isAbsolute(input.target) ? input.target : path.resolve(input.target);
+
+    // Download the file
+    const response = await fetch(input.url, { signal });
+
+    if (!response.ok) {
+      throw new Error(`Failed to download file: ${response.status} ${response.statusText}`);
+    }
+
+    // Check signal after fetch
+    if (signal?.aborted) {
+      throw new Error('Operation cancelled');
+    }
+
+    // Get the response body as array buffer
+    const arrayBuffer = await response.arrayBuffer();
```

Comment:
The file could be massive, this method could cause oom errors - stream from response to file system